src/composer.ts

```diff
@@ -370,9 +379,10 @@ export class AtomicTransactionComposer {
     }
 
     const resolvedRefIndexes: number[] = [];
-    const foreignAccounts: string[] = [];
-    const foreignApps: number[] = [];
-    const foreignAssets: number[] = [];
+    const foreignAccounts: string[] = appAccounts == null ? [] : appAccounts;
+    const foreignApps: number[] = appForeignApps == null ? [] : appForeignApps;
+    const foreignAssets: number[] =
+      appForeignAssets == null ? [] : appForeignAssets;
```

Comment:
Can you make a copy of these input arrays (e.g. `appAccounts.slice()`)?

That way we will avoid modifying inputs

Reply:
Good point, made a copy in commit: [c918bc0](https://github.com/algorand/js-algorand-sdk/pull/725/commits/c918bc0a220ed7e1fdd2c3730528f8ec240cd9a1)